chore: OPTIC-1825: Update project to use React 18

bmartel opened the pull request:
Updating the minimum required code and dependencies to support React 18. Removing a handful of redundant tests that are using a now defunct Enzyme.

### Codebase Improvements:
* Refactored `React.useEffect` hooks to avoid using async directly by defining and invoking an inner async function in `Config.jsx`, `TemplatesList.jsx`, `ModelVersionSelector.jsx`, and `BrushRegion.jsx`. [[1]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89L360-R361) [[2]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89R396-R397) [[3]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89L604-R609) [[4]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89R619-R634) [[5]](diffhunk://#diff-6fb4699dd69b2d95515e4c6e2f9f0d8a10b85e768ae4b6765330b30c2e197106L40-R50) [[6]](diffhunk://#diff-33d58198c5fa7595649983ca5e0eb3f454983be6b1134040b991dcf21f5184e1L72-R74) [[7]](diffhunk://#diff-3e1707e2a1af7817843fa466c3cbb47d7053779bb0e66be18a00168663f9fb00L514-R520) [[8]](diffhunk://#diff-3e1707e2a1af7817843fa466c3cbb47d7053779bb0e66be18a00168663f9fb00R538-R539)

### Testing and Dependencies:
* Removed Enzyme tests and related configuration in multiple test files, including `AnnotationButton.test.tsx`, `AnnotationsCarousel.test.tsx`, `Hint.test.js`, `PanelBase.test.tsx`, and `Header.test.jsx`. [[1]](diffhunk://#diff-62b08d0a6329820a4a72fd75a5ba66cb9aeb0f427c4c3072ab8f067f0e6f71f4L1-L52) [[2]](diffhunk://#diff-d856ee6f3320fe1c338af72bd059d170035cf0500c39ca38702227824d34f3dbL1-L43) [[3]](diffhunk://#diff-665f86a0d474a6bd80bd971c06bd0f92ecdf1a7a82978ed44dc80408327f86a8L1-L21) [[4]](diffhunk://#diff-1e79e0d99ebcfd7dbb20bd7fbf6f81f7b3015a7c800e5ca1da12919da9384a8eL1-L17) [[5]](diffhunk://#diff-5019e8e95ca4fa32e3acb53b32fa7fc56b2d737bafd6f58b1b62783cdc86d092L1-L102) [[6]](diffhunk://#diff-9016d799b6b3abb6b53a48b2eca0e179b76d6cf418d8829d6922373c99668fb6L6-L8) [[7]](diffhunk://#diff-9016d799b6b3abb6b53a48b2eca0e179b76d6cf418d8829d6922373c99668fb6L17-L18) [[8]](diffhunk://#diff-637daf04235c04407963ac03c54800d89e842a4586457a513643c2e1a1617189L1-L24)
* Updated `react` and `react-dom` from version 17.0.2 to 18.2.0, and removed Enzyme-related dependencies from `package.json`. [[1]](diffhunk://#diff-b861012a5dd72b8a9f3281b7cf09f5a779c98569d040b1bbc1db50f1b15e7cceL82-R86) [[2]](diffhunk://#diff-b861012a5dd72b8a9f3281b7cf09f5a779c98569d040b1bbc1db50f1b15e7cceL149-R149) [[3]](diffhunk://#diff-b861012a5dd72b8a9f3281b7cf09f5a779c98569d040b1bbc1db50f1b15e7cceL163) [[4]](diffhunk://#diff-b861012a5dd72b8a9f3281b7cf09f5a779c98569d040b1bbc1db50f1b15e7cceL175-R187) [[5]](diffhunk://#diff-b861012a5dd72b8a9f3281b7cf09f5a779c98569d040b1bbc1db50f1b15e7cceL200-R199)
* Updated `tsconfig.base.json` to format the `lib` array in a single line.

bmartel committed:
chore: Update project to use React 18

bmartel committed:
remove unused libs

bmartel committed:
remove enzyme

bmartel committed:
Merge branch 'develop' into fb-react-18

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13932589229

bmartel committed:
delete enzyme tests

bmartel committed:
Merge branch 'fb-react-18' of github.com:HumanSignal/label-studio into fb-react-18

bmartel committed:
revert change to react testing library version

AndrejOros commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13969756898)
> Docker image was pushed with the tag `20250320.125408-fb-optic-1825-38e3dab5d`

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13970413062)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1825'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13970413062

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13970413062

AndrejOros commented:
/docker build
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13970562377)
> Docker image was pushed with the tag `20250320.133323-fb-optic-1825-70d179b0d`
